src/Microsoft.DotNet.Cli.Utils/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe
+    /// </summary>
+    public static class EmbedAppNameInHost
+    {
+        private static string placeHolder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"; //hash value embedded in default apphost executable
+        private static byte[] bytesToSearch = Encoding.UTF8.GetBytes(placeHolder);
+        public static string EmbedAndReturnModifiedAppHostPath(
+            string appHostSourcePath,
+            string appHostDestinationDirectoryPath,
+            string appBinaryName)
+        {
+            var hostExtension = Path.GetExtension(appHostSourcePath);
+            var appbaseName = Path.GetFileNameWithoutExtension(appBinaryName);
+            var bytesToWrite = Encoding.UTF8.GetBytes(appBinaryName);
+            var destinationDirectory = Path.GetFullPath(appHostDestinationDirectoryPath);
+            var ModifiedAppHostPath = Path.Combine(destinationDirectory, $"{appbaseName}{hostExtension}");
+
+            if (File.Exists(ModifiedAppHostPath))
+            {
+                //We have already done the required modification to apphost.exe
+                return ModifiedAppHostPath;
+            }
+
+            if (bytesToWrite.Length > 1024)
+            {
+                throw new Exception("Strings.FileNameIsTooLong" + appBinaryName);
```

Comment:
Could we throw a better exception than this? It won't even make sense to users.

Reply:
Yes, same as I said elsewhere. I think this is just unfinished fixup from having extracted the code from sdk. Right?

Reply:
You are right. This is half done converting to a method